From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: BillyGalbreath <blake.galbreath@gmail.com>
Date: Sun, 12 Jun 2022 10:57:51 -0500
Subject: [PATCH] IceCream config files


diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
index 529fb1beadefc79ce6bf5755d5223301b9933152..badf1eaa02bc6462d134b7a912f3afd7e7404c64 100644
--- a/src/main/java/org/bukkit/Server.java
+++ b/src/main/java/org/bukkit/Server.java
@@ -2210,6 +2210,13 @@ public interface Server extends PluginMessageRecipient, net.kyori.adventure.audi
         }
         // Purpur end
 
+        // IceCream start
+        @NotNull
+        public org.bukkit.configuration.file.YamlConfiguration getIceCreamConfig() {
+            throw new UnsupportedOperationException("Not supported yet.");
+        }
+        // IceCream end
+
         /**
          * Sends the component to the player
          *
