From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pancakse <pancakseyt@gmail.com>
Date: Sat, 20 Jan 2024 11:40:28 -0700
Subject: [PATCH] IceCream Logger


diff --git a/src/main/java/org/icecream/icecream/IceCreamLogger.java b/src/main/java/org/icecream/icecream/IceCreamLogger.java
new file mode 100644
index 0000000000000000000000000000000000000000..890b91a95719f18a75bc2c2176ef5cb9f2bf4274
--- /dev/null
+++ b/src/main/java/org/icecream/icecream/IceCreamLogger.java
@@ -0,0 +1,16 @@
+package org.icecream.icecream;
+
+import org.bukkit.Bukkit;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class IceCreamLogger extends Logger {
+    public static final IceCreamLogger LOGGER = new IceCreamLogger();
+
+    private IceCreamLogger() {
+        super("IceCream", null);
+        setParent(Bukkit.getLogger());
+        setLevel(Level.ALL);
+    }
+}
